### PR TITLE
[Merged by Bors] - chore: replace remaining lambda syntax

### PR DESCRIPTION
**Pull request by grunweg**
Includes some doc comments and real code: this is exhaustive, with two exceptions:
- some files are handled in #11409 instead
- I left FunProp/{ToStd,RefinedDiscTree}.lean, Tactic/NormNum and Tactic/Simps alone, as these seem likely enough to end up in std.

Follow-up to #11301, much shorter this time.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: test/apply_fun.lean [hunk 1]
```diff
@@ -143,10 +143,8 @@ example (a b : ℕ) (h : a ≠ b) : a + 1 ≠ b + 1 := by
 -- TODO
 -- -- monotonicity will be proved by `mono` in the next example
 -- example (a b : ℕ) (h : a ≤ b) : a + 1 ≤ b + 1 :=
--- begin
---   apply_fun (λ n, n+1) at h,
+--   apply_fun (fun n ↦ n+1) at h
```
Comment (jcommelin) [inline test/apply_fun.lean:146]: Please add `by` to the preceding line

**Comment by grunweg** [comment]
Good catch, added.
awaiting-review

**Comment by eric-wieser** [comment]
bors merge

Thanks!

**Pull request closed without merging**